### PR TITLE
Added full parameter sft to llm

### PR DESCRIPTION
**Pull request by Jintao-Huang**
1. get_logger(\_\_name\_\_) -> get_logger()
2. \<function\>.\_\_get\_\_() -> MethodType
3. better PROMPT
4. Remove config['model'] = ConfigDict({...})
5. Modify the llama2 import mode
6. Add notebook mode(COMMAND_LINE_MODE=False)
7. Recommended installation from source
8. Unify get_xxx_model_tokenizer(...) -> get_model_tokenizer(model_type)
9. Modify model revision and remove sys.path.insert(...).

### CODE REVIEW

File: examples/pytorch/llm/llm_infer.py [hunk 1]
```diff
@@ -45,25 +45,17 @@ def parse_args() -> Arguments:
     return args
 
 
-args = parse_args()
+COMMAND_LINE_MODE = True
```
Comment (wenmengzhou) [inline examples/pytorch/llm/llm_infer.py:48]: write following code in a function maybe called inference,  can call it in the following code for command line mode, and in jupyter notebook , user can construct Aguments and pass it to inference by themselves

if __name__ == '__main__':
    args = parse_args()
    inference()

File: examples/pytorch/llm/llm_sft.py [hunk 2]
```diff
@@ -62,37 +67,25 @@ def parse_args() -> Arguments:
     return args
 
 
-args = parse_args()
+COMMAND_LINE_MODE = True
```
Comment (wenmengzhou) [inline examples/pytorch/llm/llm_sft.py:70]: same as llm_infer

**Review by wenmengzhou — CHANGES_REQUESTED** [review]

File: examples/pytorch/llm/llm_sft.py (hunk 2 above)
Comment (Jintao-Huang) [reply examples/pytorch/llm/llm_sft.py:70]: Ok, thank you for your advice

Comment (Jintao-Huang) [reply examples/pytorch/llm/llm_sft.py]: hello! The following error will be reported in the notebook, and I took a different approach to solve it. 

ValueError: Some specified arguments are not used by the HfArgumentParser: ['--ip=127.0.0.1', '--stdin=9008', 
'--control=9006', '--hb=9005', '--Session.signature_scheme="hmac-sha256"', 
'--Session.key=b"ee0d3388-b7c0-47e3-911b-169d49b3e834"', '--shell=9007', '--transport="tcp"', '--iopub=9009', 
'--f=/home/jintao/.local/share/jupyter/runtime/kernel-v2-109761Qud2IXHZmrdh.json']

**Comment by Jintao-Huang** [comment]
10. Added full parameter sft to llm (have tested)
11. add return_remaining_strings to HfArgumentParser
12. transformer logger to modelscope logger, and remove \_\_name\_\_
13. add save_trainer_state in BestCkptSaverHook

**Comment by Jintao-Huang** [comment]
examples of sft and infer have been tested with source installation and modelscope==1.7.2rc0

File: examples/pytorch/llm/_common.py [hunk 3]
```diff
@@ -445,7 +483,7 @@ def plot_image(tb_dir: str,
     fname = os.listdir(tb_dir)[0]
     tb_path = os.path.join(tb_dir, fname)
     data = read_tensorboard_file(tb_path)
-
+    #
```
Comment (wenmengzhou) [inline examples/pytorch/llm/_common.py:486]: remove

File: examples/pytorch/llm/llm_infer.py [hunk 4]
```diff
@@ -11,10 +11,12 @@ class Arguments:
...
             ['baichuan-7b', 'baichuan-13b', 'chatglm2', 'llama2-7b']
         })
-    ckpt_fpath: str = ''  # e.g. '/path/to/your/iter_xxx.pth'
+    sft_type: str = field(
+        default='lora', metadata={'choices': ['lora', 'full']})
+    ckpt_fpath: str = '/path/to/your/iter_xxx.pth'
     eval_human: bool = False  # False: eval test_dataset
     data_sample: Optional[int] = None
-    #
+    # sft_type: lora
```
Comment (wenmengzhou) [inline examples/pytorch/llm/llm_infer.py:19]: remove

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**